### ISSUE TITLE
Can't conpile Genfit2: No lib generated

### ISSUE THREAD

**Issue by pbielefeldt** [GenFit/GenFit#77]
Hi all!

I've (again, sorry) encountered a problem when building Genfit2. I am on *CentOS 7*, but I use a relatively recent __root 6.12/06__ and __cmake 3.11.2__. I assume I have all dependencies. I cloned the version e1dc32c of this repo created a *build* dir and used *cmake*, with the following outcome:

```
build$ cmake ..
-- The C compiler identification is GNU 4.8.5
-- The CXX compiler identification is GNU 4.8.5
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found global svn revision to be Unversioned directory
.
-- Check for ROOT_CONFIG_EXECUTABLE: /localhome/bielefeldt/allsources/build/bin/root-config
-- Check for ROOT_EXECUTABLE: /localhome/bielefeldt/allsources/build/bin/root
-- Check for ROOT_CINT_EXECUTABLE: /localhome/bielefeldt/allsources/build/bin/rootcling
-- Check for libdl.so: /usr/lib64/libdl.so
Environment variable RAVEPATH is set. Try to build GFRave.
-- Found PkgConfig: /usr/bin/pkg-config (found version "0.27.1") 
-- Checking for one of the modules 'rave'
-- Found Doxygen: /usr/bin/doxygen (found version "1.8.5") found components:  doxygen dot 
-- Could NOT find GTest (missing: GTEST_LIBRARY GTEST_INCLUDE_DIR GTEST_MAIN_LIBRARY) 
-- GTest not installed or found -- skip building unittests.
-- Configuring done
-- Generating done
-- Build files have been written to: /home/bielefeldt/Genfit2/build
```

I assume that the missing GTest is acceptable?

But with *make*, the directories __lib/__ and __bin/__ remain empty, resulting in the following:

```
build$ make
[  1%] Generating coreRootDict.cc, coreRootDict_rdict.pcm, coreRoot.rootmap
[  2%] Generating GFRaveRootDict.cc, GFRaveRootDict_rdict.pcm, GFRaveRoot.rootmap
[  3%] Generating utilitiesRootDict.cc, utilitiesRootDict_rdict.pcm, utilitiesRoot.rootmap
[  3%] Generating trackRepsRootDict.cc, trackRepsRootDict_rdict.pcm, trackRepsRoot.rootmap
[  5%] Generating measurementsRootDict.cc, measurementsRootDict_rdict.pcm, measurementsRoot.rootmap
[  6%] Generating GBLRootDict.cc, GBLRootDict_rdict.pcm, GBLRoot.rootmap
[  7%] Generating fittersRootDict.cc, fittersRootDict_rdict.pcm, fittersRoot.rootmap
[  9%] Generating eventDisplayRootDict.cc, eventDisplayRootDict_rdict.pcm, eventDisplayRoot.rootmap
Scanning dependencies of target genfit2
[ 10%] Building CXX object CMakeFiles/genfit2.dir/GFRaveRootDict.cc.o
[ 11%] Building CXX object CMakeFiles/genfit2.dir/utilitiesRootDict.cc.o
[ 12%] Building CXX object CMakeFiles/genfit2.dir/trackRepsRootDict.cc.o
[ 14%] Building CXX object CMakeFiles/genfit2.dir/measurementsRootDict.cc.o
[ 15%] Building CXX object CMakeFiles/genfit2.dir/GBLRootDict.cc.o
[ 16%] Building CXX object CMakeFiles/genfit2.dir/fittersRootDict.cc.o
[ 18%] Building CXX object CMakeFiles/genfit2.dir/eventDisplayRootDict.cc.o
[ 19%] Building CXX object CMakeFiles/genfit2.dir/coreRootDict.cc.o
In file included from /home/bielefeldt/Genfit2/core/include/AbsTrackRep.h:28:0,
                 from /home/bielefeldt/Genfit2/core/include/StateOnPlane.h:28,
                 from /home/bielefeldt/Genfit2/core/include/MeasuredStateOnPlane.h:27,
                 from /home/bielefeldt/Genfit2/core/include/MeasurementOnPlane.h:27,
                 from /home/bielefeldt/Genfit2/core/include/AbsFitterInfo.h:27,
                 from /home/bielefeldt/Genfit2/build/coreRootDict.cc:47:
/home/bielefeldt/Genfit2/core/include/Material.h:22:9: warning: unused parameter ‘material’ [-Wunused-parameter]
         Material(const Material &material) = default;
         ^
In file included from /home/bielefeldt/Genfit2/build/coreRootDict.cc:60:0:
/home/bielefeldt/Genfit2/core/include/ThinScatterer.h: In constructor ‘genfit::ThinScatterer::ThinScatterer(const SharedPlanePtr&, const genfit::Material&)’:
/home/bielefeldt/Genfit2/core/include/ThinScatterer.h:45:61: note: synthesized method ‘constexpr genfit::Material::Material(const genfit::Material&)’ first required here 
     TObject(), sharedPlane_(sharedPlane), material_(material) {;}
                                                             ^
[ 19%] Building CXX object CMakeFiles/genfit2.dir/GFRave/src/GFRaveConverters.cc.o
[ 20%] Building CXX object CMakeFiles/genfit2.dir/GFRave/src/GFRaveMagneticField.cc.o
[ 22%] Building CXX object CMakeFiles/genfit2.dir/GFRave/src/GFRavePropagator.cc.o
/home/bielefeldt/Genfit2/GFRave/src/GFRavePropagator.cc:51:1: warning: unused parameter ‘orig’ [-Wunused-parameter]
 GFRavePropagator::to ( const rave::Track & orig,
 ^
/home/bielefeldt/Genfit2/GFRave/src/GFRavePropagator.cc:51:1: warning: unused parameter ‘rcyl’ [-Wunused-parameter]
[ 23%] Building CXX object CMakeFiles/genfit2.dir/GFRave/src/GFRaveTrackParameters.cc.o
/home/bielefeldt/Genfit2/GFRave/src/GFRaveTrackParameters.cc:42:1: warning: unused parameter ‘originalState’ [-Wunused-parameter]
 GFRaveTrackParameters::GFRaveTrackParameters(const Track* track, MeasuredStateOnPlane* originalState, double weight, const TVectorD & state6, const TMatrixDSym & cov6x6, bool isSmoothed) :
 ^
/home/bielefeldt/Genfit2/GFRave/src/GFRaveTrackParameters.cc:61:1: warning: unused parameter ‘originalState’ [-Wunused-parameter]
 GFRaveTrackParameters::GFRaveTrackParameters(const Track* track, MeasuredStateOnPlane* originalState, double weight) :
 ^
[ 24%] Building CXX object CMakeFiles/genfit2.dir/GFRave/src/GFRaveVertex.cc.o
[ 25%] Building CXX object CMakeFiles/genfit2.dir/GFRave/src/GFRaveVertexFactory.cc.o
[ 27%] Building CXX object CMakeFiles/genfit2.dir/core/src/AbsFinitePlane.cc.o
[ 28%] Building CXX object CMakeFiles/genfit2.dir/core/src/AbsFitter.cc.o
[ 29%] Building CXX object CMakeFiles/genfit2.dir/core/src/AbsFitterInfo.cc.o
[ 31%] Building CXX object CMakeFiles/genfit2.dir/core/src/AbsMeasurement.cc.o
[ 32%] Building CXX object CMakeFiles/genfit2.dir/core/src/AbsTrackRep.cc.o
[ 33%] Building CXX object CMakeFiles/genfit2.dir/core/src/DetPlane.cc.o
[ 35%] Building CXX object CMakeFiles/genfit2.dir/core/src/Exception.cc.o
[ 35%] Building CXX object CMakeFiles/genfit2.dir/core/src/FieldManager.cc.o
[ 36%] Building CXX object CMakeFiles/genfit2.dir/core/src/FitStatus.cc.o
[ 37%] Building CXX object CMakeFiles/genfit2.dir/core/src/IO.cc.o
[ 38%] Building CXX object CMakeFiles/genfit2.dir/core/src/Material.cc.o
[ 40%] Building CXX object CMakeFiles/genfit2.dir/core/src/MeasuredStateOnPlane.cc.o
[ 41%] Building CXX object CMakeFiles/genfit2.dir/core/src/MeasurementOnPlane.cc.o
[ 42%] Building CXX object CMakeFiles/genfit2.dir/core/src/StateOnPlane.cc.o
[ 44%] Building CXX object CMakeFiles/genfit2.dir/core/src/ThinScatterer.cc.o
[ 45%] Building CXX object CMakeFiles/genfit2.dir/core/src/Tools.cc.o
[ 46%] Building CXX object CMakeFiles/genfit2.dir/core/src/Track.cc.o
[ 48%] Building CXX object CMakeFiles/genfit2.dir/core/src/TrackCand.cc.o
[ 49%] Building CXX object CMakeFiles/genfit2.dir/core/src/TrackCandHit.cc.o
[ 50%] Building CXX object CMakeFiles/genfit2.dir/core/src/TrackPoint.cc.o
[ 51%] Building CXX object CMakeFiles/genfit2.dir/eventDisplay/src/EventDisplay.cc.o
[ 51%] Building CXX object CMakeFiles/genfit2.dir/fields/src/ConstField.cc.o
[ 53%] Building CXX object CMakeFiles/genfit2.dir/finitePlanes/src/RectangularFinitePlane.cc.o
[ 54%] Building CXX object CMakeFiles/genfit2.dir/fitters/src/AbsKalmanFitter.cc.o
[ 55%] Building CXX object CMakeFiles/genfit2.dir/fitters/src/DAF.cc.o
[ 57%] Building CXX object CMakeFiles/genfit2.dir/fitters/src/KalmanFitStatus.cc.o
[ 58%] Building CXX object CMakeFiles/genfit2.dir/fitters/src/KalmanFittedStateOnPlane.cc.o
[ 59%] Building CXX object CMakeFiles/genfit2.dir/fitters/src/KalmanFitter.cc.o
[ 61%] Building CXX object CMakeFiles/genfit2.dir/fitters/src/KalmanFitterInfo.cc.o
[ 62%] Building CXX object CMakeFiles/genfit2.dir/fitters/src/KalmanFitterRefTrack.cc.o
[ 63%] Building CXX object CMakeFiles/genfit2.dir/fitters/src/ReferenceStateOnPlane.cc.o
[ 64%] Building CXX object CMakeFiles/genfit2.dir/GBL/src/BorderedBandMatrix.cc.o
[ 66%] Building CXX object CMakeFiles/genfit2.dir/GBL/src/GFGbl.cc.o
/home/bielefeldt/Genfit2/GBL/src/GFGbl.cc:326:6: warning: unused parameter ‘resortHits’ [-Wunused-parameter]
 void GFGbl::processTrackWithRep(Track* trk, const AbsTrackRep* rep, bool resortHits)
      ^
[ 67%] Building CXX object CMakeFiles/genfit2.dir/GBL/src/GblData.cc.o
[ 67%] Building CXX object CMakeFiles/genfit2.dir/GBL/src/GblFitter.cc.o
[ 68%] Building CXX object CMakeFiles/genfit2.dir/GBL/src/GblFitterInfo.cc.o
[ 70%] Building CXX object CMakeFiles/genfit2.dir/GBL/src/GblPoint.cc.o
[ 71%] Building CXX object CMakeFiles/genfit2.dir/GBL/src/GblTrajectory.cc.o
[ 72%] Building CXX object CMakeFiles/genfit2.dir/GBL/src/MilleBinary.cc.o
[ 74%] Building CXX object CMakeFiles/genfit2.dir/GBL/src/VMatrix.cc.o
/home/bielefeldt/Genfit2/GBL/src/VMatrix.cc: In member function ‘unsigned int gbl::VSymMatrix::invert()’:
/home/bielefeldt/Genfit2/GBL/src/VMatrix.cc:397:13: warning: declaration of ‘j’ shadows a previous local [-Wshadow]
    for (int j = 1; j <= nSize; ++j) { // elimination
             ^
/home/bielefeldt/Genfit2/GBL/src/VMatrix.cc:368:7: warning: shadowed declaration is here [-Wshadow]
   int j = first;
       ^
/home/bielefeldt/Genfit2/GBL/src/VMatrix.cc:434:13: warning: declaration of ‘k’ shadows a previous local [-Wshadow]
    for (int k = 1; k <= nSize; ++k) {
             ^
/home/bielefeldt/Genfit2/GBL/src/VMatrix.cc:365:7: warning: shadowed declaration is here [-Wshadow]
   int k = 0;
       ^
/home/bielefeldt/Genfit2/GBL/src/VMatrix.cc:437:15: warning: declaration of ‘j’ shadows a previous local [-Wshadow]
      for (int j = 1; j <= nSize; ++j) {
               ^
/home/bielefeldt/Genfit2/GBL/src/VMatrix.cc:368:7: warning: shadowed declaration is here [-Wshadow]
   int j = first;
       ^
[ 75%] Building CXX object CMakeFiles/genfit2.dir/measurements/src/FullMeasurement.cc.o
[ 76%] Building CXX object CMakeFiles/genfit2.dir/measurements/src/HMatrixPhi.cc.o
[ 77%] Building CXX object CMakeFiles/genfit2.dir/measurements/src/HMatrixU.cc.o
[ 79%] Building CXX object CMakeFiles/genfit2.dir/measurements/src/HMatrixUV.cc.o
[ 80%] Building CXX object CMakeFiles/genfit2.dir/measurements/src/HMatrixUnit.cc.o
[ 81%] Building CXX object CMakeFiles/genfit2.dir/measurements/src/HMatrixV.cc.o
[ 83%] Building CXX object CMakeFiles/genfit2.dir/measurements/src/PlanarMeasurement.cc.o
[ 84%] Building CXX object CMakeFiles/genfit2.dir/measurements/src/ProlateSpacepointMeasurement.cc.o
[ 84%] Building CXX object CMakeFiles/genfit2.dir/measurements/src/SpacepointMeasurement.cc.o
[ 85%] Building CXX object CMakeFiles/genfit2.dir/measurements/src/WireMeasurement.cc.o
[ 87%] Building CXX object CMakeFiles/genfit2.dir/measurements/src/WireMeasurementNew.cc.o
[ 88%] Building CXX object CMakeFiles/genfit2.dir/measurements/src/WirePointMeasurement.cc.o
[ 89%] Building CXX object CMakeFiles/genfit2.dir/measurements/src/WireTrackCandHit.cc.o
[ 90%] Building CXX object CMakeFiles/genfit2.dir/trackReps/src/MaterialEffects.cc.o
[ 92%] Building CXX object CMakeFiles/genfit2.dir/trackReps/src/RKTools.cc.o
[ 93%] Building CXX object CMakeFiles/genfit2.dir/trackReps/src/RKTrackRep.cc.o
[ 94%] Building CXX object CMakeFiles/genfit2.dir/trackReps/src/StepLimits.cc.o
[ 96%] Building CXX object CMakeFiles/genfit2.dir/trackReps/src/TGeoMaterialInterface.cc.o
[ 97%] Building CXX object CMakeFiles/genfit2.dir/utilities/src/HelixTrackModel.cc.o
[ 98%] Building CXX object CMakeFiles/genfit2.dir/utilities/src/MeasurementCreator.cc.o
[100%] Linking CXX shared library lib/libgenfit2.so
c++: error: missing path after ‘-I’
make[2]: *** [lib/libgenfit2.so.2.2.0] Error 1
make[1]: *** [CMakeFiles/genfit2.dir/all] Error 2
make: *** [all] Error 2
```

Am I doing something wrong? I basically followed your instructions/the travis config. Is there maybe a problem with the provided CMakeList?

**Comment by pbielefeldt** [comment]
Hm ... It only happens to me if I have my environment variable `RAVEPATH` set. Is *rave* still used for Genfit? At least the travis script does not mention it any longer (that's probably the reason a set RAVEPATH env might break things without anyone taking notice). On the other hand, *CMake* will issue a warning if you run it without RAVEPATH being set ...?  
If rave doesn't do anything but creating warnings and compile errors, I propose to remove it from the CMakeList ;-) But: Right now, the result of `./bin/unitTests` is:

```
name                            	pass 	fail 	exception
checkSetGetPosMom()             	1000 	0 	0
compareForthBackExtrapolation() 	578 	50 	372
checkStopAtBoundary()           	1000 	0 	0
checkErrorPropagation()         	1000 	0 	0
checkExtrapolateToLine()        	1000 	0 	0
checkExtrapolateToPoint()       	1000 	0 	0
checkExtrapolateToCylinder()    	579 	0 	421
checkExtrapolateToSphere()      	118 	0 	882
checkExtrapolateBy()            	1000 	0 	0
compareForthBackJacNoise()      	975 	25 	0
```

I don't know whether the failed tests are because of rave missing -- and since Genfit2 doesn't compile on my machines with `RAVEPATH` set, I can't check how its tests would look like with rave installed, sorry.

**Comment by pinkenburg** [comment]
sPHENIX is using it

**Comment by nils-braun** [comment]
Sorry for not answering, I think @pinkenburg has solved this perfectly with #81

**Comment by MarkusPrim** [comment]
Closing based on @nils-braun comment. If it needs to be re-opened please do so.
